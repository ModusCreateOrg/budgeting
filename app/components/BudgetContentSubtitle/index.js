// @flow
import * as React from 'react';
import formatPercent from 'utils/formatPercent';
import styles from './style.scss';

type BudgetContentSubtitleProps = {
  contributionPercentage: object,
};

const BudgetContentSubtitle = ({ contributionPercentage }: BudgetContentSubtitleProps) => (
  <h2>
    {contributionPercentage.value > 0 && <span className={styles.pos}>+</span>}
    {contributionPercentage.value < 0 && <span className={styles.neg}>-</span>}
    {formatPercent(contributionPercentage.contribution, false).text}
  </h2>
);

export default BudgetContentSubtitle;
