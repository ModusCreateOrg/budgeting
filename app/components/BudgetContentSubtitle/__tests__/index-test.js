import React from 'react';
import renderer from 'react-test-renderer';
import BudgetContentSubtitle from 'components/BudgetContentSubtitle';

it('renders correctly', () => {
  const contributionPercentage = {
    value: 100,
    contribution: 10,
  };

  const tree = renderer.create(<BudgetContentSubtitle contributionPercentage={contributionPercentage} />).toJSON();
  expect(tree).toMatchSnapshot();
});
