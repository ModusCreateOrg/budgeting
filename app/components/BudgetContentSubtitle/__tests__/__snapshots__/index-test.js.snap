// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<h2>
  <span
    className="pos"
  >
    +
  </span>
  1,000%
</h2>
`;
